xbmc/pvr/timers/PVRTimers.cpp

```diff
@@ -1155,7 +1155,8 @@ std::shared_ptr<CPVRTimerInfoTag> CPVRTimers::GetTimerForEpgTag(const std::share
         if (timersEntry->GetEpgInfoTag(false) == epgTag)
           return timersEntry;
 
-        if (timersEntry->m_iClientChannelUid != PVR_CHANNEL_INVALID_UID &&
+        if (timersEntry->m_iClientId == epgTag->ClientID() &&
+            timersEntry->m_iClientChannelUid != PVR_CHANNEL_INVALID_UID &&
             timersEntry->m_iClientChannelUid == epgTag->UniqueChannelID())
```

Comment:
From my side OK, maybe only to make this new compare as last? The work about `timersEntry->m_iClientChannelUid != PVR_CHANNEL_INVALID_UID` is smaller (no function call to get value) and to stop if this false.
```cpp
        if (timersEntry->m_iClientChannelUid != PVR_CHANNEL_INVALID_UID &&
            timersEntry->m_iClientChannelUid == epgTag->UniqueChannelID() &&
            timersEntry->m_iClientId == epgTag->ClientID())
```